src/event-bus/EventBus.ts

```diff
@@ -0,0 +1,41 @@
+export type Subscriber = (message: any) => void;
+export interface Message {
+    key?: any;
+    payload: any;
+}
+
+/** A topic-based event bus to facilitate communication between plugins */
+export default class EventBus {
```

Comment:
Sorry, I renamed it to dispatch since notify was a bad name.

Notify is usually to notify of state changes that have already happened. Dispatch is usually for sending messages. So I thought dispatch was a better name